### PR TITLE
Allow uers to specify the name of moving mean and variance in batch_norm interface.

### PR DESCRIPTION
**Pull request by qingqing01**
Fix https://github.com/PaddlePaddle/Paddle/issues/8068

### CODE REVIEW

File: python/paddle/v2/fluid/layers/nn.py [hunk 1]
```diff
@@ -1508,13 +1510,15 @@ def batch_norm(input,
         attr=helper.bias_attr, shape=param_shape, dtype=dtype, is_bias=True)
 
     mean = helper.create_global_variable(
+        name=helper.moving_mean_name,
```
Comment (chengduoZH) [inline python/paddle/v2/fluid/layers/nn.py]: `helper.moving_mean_name` ==> `moving_mean_name`

`helper` doesn't have this property.
https://github.com/PaddlePaddle/Paddle/blob/8894c67d7168dfa5f5dc8e57ec2b5c60f24e368c/python/paddle/v2/fluid/layer_helper.py#L24-L107

Comment (qingqing01) [reply python/paddle/v2/fluid/layers/nn.py]: @chengduoZH 

In line 1488:

```python
helper = LayerHelper('batch_norm', **locals())
```

```python
 class LayerHelper(object): 
     def __init__(self, layer_type, **kwargs): 
         self.kwargs = kwargs 
```

So helper.moving_mean_name is no problem.

Comment (qingqing01) [reply python/paddle/v2/fluid/layers/nn.py]: @chengduoZH Done. change `helper.moving_mean_name` to `moving_mean_name`

File: python/paddle/v2/fluid/layers/nn.py [hunk 2]
```diff
@@ -1478,7 +1478,9 @@ def batch_norm(input,
                param_attr=None,
                bias_attr=None,
                data_layout='NCHW',
-               name=None):
+               name=None,
```
Comment (jacquesqiao) [inline python/paddle/v2/fluid/layers/nn.py]: if this `name` is still used?

**Pull request merged**